README.md

```diff
@@ -0,0 +1,560 @@
+- The `onResume()` and `onPause()` functions are ensuring that the message handler is registered while the device is in these states. 
+- While the application is in the foreground the `onMessage()` function is called to handle the push notification. In this example its a simple toast that is shown to the user. 
+- Finally the `toastStartUpPushNotification()` is an example that is called when the user launches the application. If the application is launched by the user clicking in the push notification, that notification gets toasted to the user.
+
+```java
+public class MainActivity extends AppCompatActivity implements MessageHandler {
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        
+        setupPush();
+        toastStartUpPushNotification();
+    }
+    
+    @Override
+    protected void onResume() {
+        super.onResume();
+        RegistrarManager.registerMainThreadHandler(this);
+        RegistrarManager.unregisterBackgroundThreadHandler(NotifyingHandler.instance);
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        RegistrarManager.unregisterMainThreadHandler(this);
+        RegistrarManager.registerBackgroundThreadHandler(NotifyingHandler.instance);
+    }
+
+    @Override
+    public void onMessage(Context context, Bundle bundle) {
+        // display the message contained in the payload
+        Toast.makeText(getApplicationContext(),
+                bundle.getString(UnifiedPushMessage.ALERT_KEY), Toast.LENGTH_LONG).show();
+
+    }
+    
+    private void setupPush(){
+        PushApplication application = (PushApplication) getApplication();
+        PushRegistrar pushRegistrar = application.getPushRegistrar();
+        pushRegistrar.register(getApplicationContext(), new Callback<Void>() {
+            
+            @Override
+            public void onSuccess(Void data) {
+                Log.d(TAG, "Registration Succeeded");
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                Log.e(TAG, e.getMessage(), e);
+                Toast.makeText(getApplicationContext(),
+                        "Ops, something is wrong :(", Toast.LENGTH_LONG).show();
+            }
+        });
+    }
+    
+    public void toastStartUpPushNotification(){
+        if (getIntent() != null) {
+            Bundle bundle = getIntent().getExtras();
+            if (bundle != null && bundle.getString(UnifiedPushMessage.ALERT_KEY) != null) {
+                Toast.makeText(getApplicationContext(),
+                        bundle.getString(UnifiedPushMessage.ALERT_KEY), Toast.LENGTH_LONG).show();
+            }
+
+        }
+    }
+
+}
+```
+More information can be found at [docs.aerogear.org](https://docs.aerogear.org/aerogear/latest/push-notifications.html).
+
+### 4. Mobile-services parser
```

Comment:
It's very likely that we may get rid of this in the future. So I suggest that we don't reduce the content in this section here. Most of the things can be code comments (which can be used to generate javadoc), and will be removed when the code is removed.